### PR TITLE
Fixed broken links and unused references.

### PR DESCRIPTION
**Pull request by abhipsaMisra**
# Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [x] This pull-request is submitted against the `master` branch.

# Description of the changes

Fixed broken links and cleans unused references in the readme files

# Reference/Link to the issue solved with this PR (if any)

### CODE REVIEW

File: doc/faq.md [hunk 1]
```diff
@@ -1,160 +1,58 @@
 # Microsoft Azure IoT device SDK FAQ
```
Comment (CIPop) [inline doc/faq.md]: @abhipsaMisra I remember we've agreed to delete this. UWP support is partially implemented.

File: doc/manage_iot_hub.md [hunk 2]
```diff
@@ -4,105 +4,28 @@ Before a device can communicate with IoT Hub, you must add details of that devic
 
 To add devices to your IoT hub and manage those devices, you can use either of:
```
Comment (CIPop) [inline doc/manage_iot_hub.md]: @abhipsaMisra I remember we've agreed to remove this.
We have better MSDN documentation for IoT Hub and DPS management.

File: doc/setup_iothub.md [hunk 3]
```diff
@@ -6,55 +6,12 @@ Before you can communicate with IoT Hub from a device you must create an IoT hub
 
 ## Create an IoT hub
```
Comment (CIPop) [inline doc/setup_iothub.md]: Same as above: this should be a document on MSDN.

File: iothub/device/readme.md [hunk 4]
```diff
@@ -15,8 +15,6 @@ The library is available as a NuGet package for you include in your own developm
 ## Application development guidelines
```
Comment (CIPop) [inline iothub/device/readme.md]: Same: please remove - application development does not require setting up a dev environment to work on SDKs. Instead NuGet packages can be used.

File: doc/manage_iot_hub.md (hunk 2 above)
Comment (abhipsaMisra) [reply doc/manage_iot_hub.md]: The text on the readme redirects to the MSDN document on Iot Hub and DPS

File: doc/setup_iothub.md (hunk 3 above)
Comment (abhipsaMisra) [reply doc/setup_iothub.md]: The text on the readme redirects to the MSDN document on Iot Hub and DPS

File: iothub/device/readme.md (hunk 4 above)
Comment (abhipsaMisra) [reply iothub/device/readme.md]: updated

**Pull request merged**